http-clients/apache-client/src/main/java/software/amazon/awssdk/http/apache/ApacheHttpClient.java

```diff
@@ -403,6 +403,11 @@ public interface Builder extends SdkHttpClient.Builder<ApacheHttpClient.Builder>
          */
         Builder dnsResolver(DnsResolver dnsResolver);
 
+        /**
+         * Configuration that defines a Socket Factory. If no matches are found, the default factory is used.
+         */
+        Builder socketFactory(ConnectionSocketFactory socketFactory);
```

Comment:
Are there any other pieces of configuration that are not used if this socket factory is set? E.g. TRUST_ALL_CERTIFICATES? Can we document what those are?

Reply:
Added paragraph with three options that will get ignored